app-common/src/lib.rs

```diff
@@ -0,0 +1,246 @@
+    Config { message: String },
+}
+
+
+#[derive(Args, Debug)]
+pub struct TLSOptions {
+    /// Enables mTLS (TLS for DICOM connections)
+    #[arg(long = "tls", default_value = "false")]
+    pub enabled: bool,
+
+    /// Crypto provider to use, see documentation (https://docs.rs/rustls/latest/rustls/index.html) for details
+    #[arg(long, value_enum, default_value_t = CryptoProvider::AwsLC, value_name = "provider")]
+    pub crypto_provider: CryptoProvider,
+
+    /// List of cipher suites to use. If not specified, the default cipher suites for the selected crypto provider will be used.
+    #[arg(long, value_name = "cipher1,...")]
+    pub cipher_suites: Option<Vec<String>>,
+
+    /// TLS protocol versions to enable
+    #[arg(long, value_enum, value_name = "version,...", default_values_t = vec![TLSProtocolVersion::TLS1_2, TLSProtocolVersion::TLS1_3])]
+    pub protocol_versions: Vec<TLSProtocolVersion>,
+
+    /// Path to private key file in PEM format
+    #[arg(long, value_name = "/path/to/key.pem,...")]
+    pub key: Option<PathBuf>,
+
+    /// Path to certificate file in PEM format
+    #[arg(long, value_name = "/path/to/cert.pem,...")]
+    pub cert: Option<PathBuf>,
+
+    /// Path to additional CA certificates (comma separated) in PEM format to add to the root store
+    #[arg(long, value_name = "/path/to/cert.pem,...")]
+    pub add_certs: Option<Vec<PathBuf>>,
+
+    /// Add Certificate Revocation Lists (CRLs) to the server's certificate verifier
+    #[arg(long, value_name = "/path/to/crl.pem,...")]
+    pub add_crls: Option<Vec<PathBuf>>,
+
+    #[arg(long, action = clap::ArgAction::SetFalse)]
+    /// Load certitificates from the system root store
+    pub system_roots: bool,
+
+    /// How to handle peer certificates
+    #[arg(long, value_enum, value_name = "opt", default_value_t = PeerCertOption::Require)]
+    pub peer_cert: PeerCertOption,
+
+    /// Allow unauthenticated clients (only valid for server)
+    #[arg(long)]
+    pub allow_unauthenticated: bool,
+}
+
+/// Crypto provider options
+/// 
+/// See rustls 
+/// [Cryptograpy providers](https://docs.rs/rustls/latest/rustls/#cryptography-providers)
+/// for more details
+/// 
+/// Currently only AWS-LC is supported
+#[derive(clap::ValueEnum, Clone, Debug)]
+pub enum CryptoProvider {
+    AwsLC,
+    //RING
+}
+
+/// TLS protocol version options
+/// 
+/// Subset of rustls 
+/// [ProtocolVersions](https://docs.rs/rustls/latest/rustls/enum.ProtocolVersion.html#variants) 
+/// supported
+#[derive(clap::ValueEnum, Clone, Debug)]
+pub enum TLSProtocolVersion {
+    TLS1_2,
+    TLS1_3,
+}
+
+/// Peer certificate handling options
+/// 
+/// Defines how the TLS connection should handle peer certificates
+#[derive(clap::ValueEnum, Clone, Debug)]
+pub enum PeerCertOption {
+    /// Require the peer to present a valid certificate
+    Require,
+    /// Do not verify the peer certificate
+    Ignore,
+}
+
+pub fn show_cipher_suites() -> ! {
```

Comment:
Though it makes sense for command-line applications to exit after showing the list of supported cipher suites, perhaps we can delegate that to the end application and restrict this function to printing?